server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -95,6 +101,32 @@ private NLPResult analyzeWithNLP(String text) throws IOException {
     return nlpResults;
   }
 
+  /**
+   * Takes the given map of sentiments and creates a common weighted 
+   * sentiment according to the number of likes each comment receives
+   * out of the sum of the likes of the comments
+   *
+   * @param unweightedNLPMap the unweighted map of <like, nlp> pairs
+   * @return A weighted NLPResult object
+   */
+  private NLPResult createWeightedSentiment(HashMap<Integer, NLPResult> unweightedNLPMap) {
+      double totalLikes = 0;
+
+      for (Map.Entry<Integer, NLPResult> likesAndNLPResult : unweightedNLPMap.entrySet()) {
+          totalLikes += (double) likesAndNLPResult.getKey();
+      }
+
+      double weightedMagnitude = 0;
+      double weightedScore = 0;
+
+      for (Map.Entry<Integer, NLPResult> likesAndNLPResult : unweightedNLPMap.entrySet()) {
+          weightedMagnitude += (likesAndNLPResult.getKey() / totalLikes) * likesAndNLPResult.getValue().magnitude;
+          weightedScore += (likesAndNLPResult.getKey() / totalLikes) * likesAndNLPResult.getValue().score;
```

Comment:
Shouldn't you just weigh magnitude? Weighing the score means that you change the meaning of the sentiment based on number of comments, which seems incorrect.

Reply:
The score is how negative/positive a comment is, while the magnitude is how confident NLP is that it's correct basically. 

Here i'm saying that, to determine how negative/postiive the entire group is, I'm taking how negative/positive each comment is and proportionally adding it to a total magnitude and score value that reflects both the negative/positive-ness of a group as well as the confidence that group is negative/positive

Reply:
Ah, then we should only be modifying the score and not the magnitude then.

Reply:
My only case is how will I figure out the correct magnitude for each of the 20 comments then? If a comment which has a high number of likes has a high magnitude or "correctness", shouldn't that "correctness" also be weighted?

For example, if there's a comment that is is clearly positive, and gets 100k likes, shouldn't that clearness be taken into account more than a comment that is maybe positive and only gets 10k likes?

Reply:
Spoke offline. The magnitude should be weighted since we're trying to determine how confident we are on our overall score, so we want to give higher weight to our confidence in our scores that count for more.